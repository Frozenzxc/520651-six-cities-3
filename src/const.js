const OfferType = {
  ALL: `cities`,
  NEARBY: `near`,
};

const SortType = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  RATING: `Top rated first`,
};

const MAX_NEARBY_OFFERS = 3;

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export {OfferType, SortType, MAX_NEARBY_OFFERS, AuthorizationStatus};
