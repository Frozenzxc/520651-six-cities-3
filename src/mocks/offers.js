import {getRandomArrayItem, getRandomIntegerNumber} from "../common";

const MAX_BEDROOMS = 6;
const MIN_BEDROOMS = 1;
const MAX_GUESTS = 10;
const MAX_PHOTOS = 6;
const MAX_RATING = 5;
const MIN_RATING = 1;

const CardTypes = [`apartment`, `room`, `house`, `hotel`];
const Options = [`Wifi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`];
const Photos = [
  `img/apartment-01.jpg`,
  `img/apartment-02.jpg`,
  `img/apartment-03.jpg`,
  `img/apartment-01.jpg`,
  `img/apartment-02.jpg`,
  `img/apartment-03.jpg`,
];

const generatePhotos = (photos) => {
  return photos
        .filter(() => Math.random() > 0.5)
        .slice(0, getRandomIntegerNumber(1, MAX_PHOTOS));
};

const generateOptions = (options) => {
  return options
        .filter(() => Math.random() > 0.5)
        .slice(0, getRandomIntegerNumber(0, Options.length));
};

export default [
  {
    bedroomsCount: 4,
    coordinates: [52.3909553943508, 4.85309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      name: `John`,
      avatar: `img/avatar-max.jpg`,
      super: false,
    },
    id: 1,
    maxGuests: 6,
    options: [`Wifi`, `Washing machine`, `Towels`, `Heating`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    premium: false,
    price: 200,
    rating: 3,
    src: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious apartment at great location`,
    type: `house`,
  },
  {
    bedroomsCount: 2,
    coordinates: [52.369553943508, 4.85309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      name: `Max`,
      avatar: `img/avatar-angelina.jpg`,
      super: false,
    },
    id: 2,
    maxGuests: 4,
    options: [`Wifi`, `Towels`, `Heating`, `Cabel TV`, `Fridge`],
    premium: true,
    price: 300,
    rating: 1,
    src: [`img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Wood and stone place`,
    type: `hotel`,
  },
  {
    bedroomsCount: 1,
    coordinates: [52.3909553943508, 4.929309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      name: `Bob`,
      avatar: `img/avatar-max.jpg`,
      super: true,
    },
    id: 3,
    maxGuests: 2,
    options: [`Wifi`, `Towels`, `Cabel TV`],
    premium: false,
    price: 450,
    rating: 3,
    src: [`img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Canal View Prinsengracht`,
    type: `room`,
  },
  {
    bedroomsCount: 2,
    coordinates: [52.3809553943508, 4.939309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      super: false,
    },
    id: 4,
    maxGuests: 5,
    options: [`Wifi`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Cabel TV`],
    premium: false,
    price: 150,
    rating: 5,
    src: [`img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Nice, cozy, warm big bed apartment`,
    type: `apartment`,
  },
];
