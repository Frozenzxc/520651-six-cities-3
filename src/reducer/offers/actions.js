const ActionType = {
  BLOCK_FORM: `BLOCK_FORM`,
  LOAD_NEARBY_OFFERS: `LOAD_NEARBY_OFFERS`,
  LOAD_OFFERS: `LOAD_OFFERS`,
  LOAD_REVIEWS: `LOAD_REVIEWS`,
  SELECT_CARD: `SELECT_CARD`,
  SELECT_CITY: `SELECT_CITY`,
  SELECT_OFFER: `SELECT_OFFER`,
  SUCCESSFUL_POST_REVIEW: `SUCCESSFUL_POST_REVIEW`,
};

const ActionCreator = {

  blockForm: (isBlocked) => {
    return {
      type: ActionType.BLOCK_FORM,
      payload: isBlocked,
    };
  },

  loadNearbyOffers: (offers) => {
    return {
      type: ActionType.LOAD_NEARBY_OFFERS,
      payload: offers,
    };
  },

  loadOffers: (offers) => {
    return {
      type: ActionType.LOAD_OFFERS,
      payload: offers,
    };
  },

  loadReviews: (reviews) => {
    return {
      type: ActionType.LOAD_REVIEWS,
      payload: reviews,
    };
  },

  selectCard: (offer) => ({
    type: ActionType.SELECT_CARD,
    payload: offer.id,
  }),

  selectCity: (city) => ({
    type: ActionType.SELECT_CITY,
    payload: city,
  }),

  selectOffer: (offer) => ({
    type: ActionType.SELECT_OFFER,
    payload: offer,
  }),

  successfulPostReview: (isSuccess) => ({
    type: ActionType.SUCCESSFUL_POST_REVIEW,
    payload: isSuccess,
  })

};

export {ActionType, ActionCreator};
