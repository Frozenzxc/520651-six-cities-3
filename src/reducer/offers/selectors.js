import {createSelector} from "reselect";
import NameSpace from "../name-space";
import {MAX_NEARBY_OFFERS} from "../../const";

const getCurrentCity = (state) => {
  return state[NameSpace.OFFERS].currentCity;
};

const getActiveOffer = (state) => {
  return state[NameSpace.OFFERS].activeOffer;
};

const getFavoriteOffers = (state) => {
  return state[NameSpace.OFFERS].favoriteOffers;
};

const getNearbyOffers = (state) => {
  return state[NameSpace.OFFERS].nearbyOffers.slice(0, MAX_NEARBY_OFFERS);
};

const getOffers = (state) => {
  return state[NameSpace.OFFERS].offers;
};

const getReviews = (state) => {
  return state[NameSpace.OFFERS].reviews;
};

const getAvailableOffers = createSelector(
    getOffers,
    getCurrentCity,
    (offers, currentCity) => {
      return offers.filter((offer) => offer.city.name === currentCity);
    }
);

export {getCurrentCity, getActiveOffer, getNearbyOffers, getOffers, getReviews, getAvailableOffers, getFavoriteOffers};
