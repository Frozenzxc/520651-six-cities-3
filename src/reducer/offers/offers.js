import {extend} from "../../utils.js";
import {parseOffer, parseReview} from "../../utils";
import {ActionCreator, ActionType} from "../offers/actions";
import {ReviewPostingStatus} from "../../const";

const getAvailableOffers = ((allOffers, currentCity) => allOffers.filter((offer) => offer.city.name === currentCity));

const initialState = {
  offers: [],
  availableOffers: [],
  activeID: null,
  activeOffer: null,
  currentCity: `Amsterdam`,
  isFormBlocked: false,
  isLoading: true,
  isPropertyLoading: true,
  nearbyOffers: [],
  reviews: [],
  reviewPostingStatus: null,
};

const Operation = {
  loadNearbyOffers: (id) => (dispatch, getState, api) => {
    return api.get(`/hotels/${id}/nearby`)
            .then((response) => {
              dispatch(ActionCreator.loadNearbyOffers(response.data));
            });
  },

  loadOffers: () => (dispatch, getState, api) => {
    return api.get(`/hotels`)
            .then((response) => {
              dispatch(ActionCreator.loadOffers(response.data));
            });
  },

  loadReviews: (id) => (dispatch, getState, api) => {
    return api.get(`/comments/${id}`)
            .then((response) => {
              dispatch(ActionCreator.loadReviews(response.data));
            });
  },

  postReview: (id, formData) => (dispatch, getState, api) => {
    return api.post(`/comments/${id}`, {
      comment: formData.comment,
      rating: formData.rating,
    })
          .then((response) => {
            dispatch(ActionCreator.successfulPostReview(ReviewPostingStatus.POSTED));
            dispatch(ActionCreator.loadReviews(response.data));
            dispatch(ActionCreator.blockForm(false));
          })
        .catch((err) => {
          dispatch(ActionCreator.successfulPostReview(ReviewPostingStatus.ERROR));
          dispatch(ActionCreator.blockForm(false));
          throw err;
        });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.BLOCK_FORM:
      return extend(state, {
        isFormBlocked: action.payload,
      });

    case ActionType.LOAD_NEARBY_OFFERS:
      let parsedNearbyOffers = action.payload.map((offer) => parseOffer(offer));
      return extend(state, {
        nearbyOffers: parsedNearbyOffers,
      });

    case ActionType.LOAD_OFFERS:
      let parsedOffers = action.payload.map((offer) => parseOffer(offer));
      return extend(state, {
        currentCity: parsedOffers[0].city.name,
        isLoading: false,
        offers: parsedOffers,
      });

    case ActionType.LOAD_REVIEWS:
      let parsedReviews = action.payload.map((review) => parseReview(review));
      return extend(state, {
        isPropertyLoading: false,
        reviews: parsedReviews,
      });

    case ActionType.SELECT_CARD:
      return extend(state, {
        activeID: action.payload,
      });

    case ActionType.SELECT_CITY:
      return extend(state, {
        currentCity: action.payload,
        availableOffers: getAvailableOffers(state.offers, action.payload),
      });

    case ActionType.SELECT_OFFER:
      return extend(state, {
        activeOffer: action.payload,
      });

    case ActionType.SUCCESSFUL_POST_REVIEW:
      return extend(state, {
        reviewPostingStatus: action.payload,
      });
  }

  return state;
};

export {reducer, Operation, ActionType, ActionCreator};
