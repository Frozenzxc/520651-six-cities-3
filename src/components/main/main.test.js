import React from "react";
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import Main from "./main.jsx";
import offers from "../../test-mocks/test-offers";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";
import {AuthorizationStatus} from "../../const";

const mockStore = configureStore([]);

it(`Should Main component render correctly with not authorized user`, () => {

  const store = mockStore({
    [NameSpace.OFFERS]: {
      availableOffers: offers.filter((offer) => offer.city.name === offers[0].city.name),
      offers,
      currentCity: offers[0].city.name,
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      authEmail: null,
    },
  });

  const tree = renderer
        .create(
            <Provider store={store}>
              <BrowserRouter>
                <Main
                  activeOffer={offers[0]}
                  availableOffers={offers.filter((offer) => offer.city.name === offers[0].city.name)}
                  authEmail={null}
                  authorizationStatus={AuthorizationStatus.NO_AUTH}
                  onCardHover={() => {}}
                  onCardTitleClick={() => {}}
                  currentCity={offers[0].city.name}
                />
              </BrowserRouter>
            </Provider>
        )
        .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Should Main component render correctly with authorized user`, () => {

  const store = mockStore({
    [NameSpace.OFFERS]: {
      availableOffers: offers.filter((offer) => offer.city.name === offers[0].city.name),
      offers,
      currentCity: offers[0].city.name,
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
      authEmail: `AAA@adfg.ru`,
    },
  });

  const tree = renderer
        .create(
            <Provider store={store}>
              <BrowserRouter>
                <Main
                  activeOffer={offers[0]}
                  availableOffers={offers.filter((offer) => offer.city.name === offers[0].city.name)}
                  authEmail={`AAA@adfg.ru`}
                  authorizationStatus={AuthorizationStatus.AUTH}
                  onCardHover={() => {}}
                  onCardTitleClick={() => {}}
                  currentCity={offers[0].city.name}
                />
              </BrowserRouter>
            </Provider>
        )
        .toJSON();

  expect(tree).toMatchSnapshot();
});
