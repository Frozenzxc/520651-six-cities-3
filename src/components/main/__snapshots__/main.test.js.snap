// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Main component render correctly with authorized user 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  AAA@adfg.ru
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              data-city="Cologne"
              href="#"
            >
              <span
                data-city="Cologne"
              >
                Cologne
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item "
              data-city="Paris"
              href="#"
            >
              <span
                data-city="Paris"
              >
                Paris
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item "
              data-city="Hamburg"
              href="#"
            >
              <span
                data-city="Hamburg"
              >
                Hamburg
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item "
              data-city="Brussels"
              href="#"
            >
              <span
                data-city="Brussels"
              >
                Brussels
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            3
             places to stay in 
            Cologne
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom false"
            >
              <li
                className="places__option places__option--active"
                data-sort-type="Popular"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option false"
                data-sort-type="Price: low to high"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option false"
                data-sort-type="Price: high to low"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option false"
                data-sort-type="Top rated first"
                onClick={[Function]}
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="cities-places__list places__list tabs__content"
          >
            <article
              className="cities-places__card place-card"
              onMouseEnter={[Function]}
            >
              <div
                className="cities-image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      160
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "96%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                  onClick={[Function]}
                >
                  <a
                    href="#"
                  >
                    The house among olive 
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  apartment
                </p>
              </div>
            </article>
            <article
              className="cities-places__card place-card"
              onMouseEnter={[Function]}
            >
              <div
                className="cities-image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      192
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "84%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                  onClick={[Function]}
                >
                  <a
                    href="#"
                  >
                    Waterfront with extraordinary view
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  room
                </p>
              </div>
            </article>
            <article
              className="cities-places__card place-card"
              onMouseEnter={[Function]}
            >
              <div
                className="cities-image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      196
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "66%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                  onClick={[Function]}
                >
                  <a
                    href="#"
                  >
                    Loft Studio in the Central Area
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  house
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <div
              id="map"
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`Should Main component render correctly with not authorized user 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <span
                  className="header__login"
                  onClick={[Function]}
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              data-city="Cologne"
              href="#"
            >
              <span
                data-city="Cologne"
              >
                Cologne
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item "
              data-city="Paris"
              href="#"
            >
              <span
                data-city="Paris"
              >
                Paris
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item "
              data-city="Hamburg"
              href="#"
            >
              <span
                data-city="Hamburg"
              >
                Hamburg
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item "
              data-city="Brussels"
              href="#"
            >
              <span
                data-city="Brussels"
              >
                Brussels
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            3
             places to stay in 
            Cologne
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom false"
            >
              <li
                className="places__option places__option--active"
                data-sort-type="Popular"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option false"
                data-sort-type="Price: low to high"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option false"
                data-sort-type="Price: high to low"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option false"
                data-sort-type="Top rated first"
                onClick={[Function]}
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="cities-places__list places__list tabs__content"
          >
            <article
              className="cities-places__card place-card"
              onMouseEnter={[Function]}
            >
              <div
                className="cities-image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      160
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "96%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                  onClick={[Function]}
                >
                  <a
                    href="#"
                  >
                    The house among olive 
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  apartment
                </p>
              </div>
            </article>
            <article
              className="cities-places__card place-card"
              onMouseEnter={[Function]}
            >
              <div
                className="cities-image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      192
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "84%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                  onClick={[Function]}
                >
                  <a
                    href="#"
                  >
                    Waterfront with extraordinary view
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  room
                </p>
              </div>
            </article>
            <article
              className="cities-places__card place-card"
              onMouseEnter={[Function]}
            >
              <div
                className="cities-image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      196
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "66%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                  onClick={[Function]}
                >
                  <a
                    href="#"
                  >
                    Loft Studio in the Central Area
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  house
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <div
              id="map"
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
